src/code/z_en_item00.c

```diff
@@ -732,36 +738,37 @@ void EnItem00_Update(Actor* thisx, GlobalContext* globalCtx) {
     }
 
     this->actionFunc(this, globalCtx);
-
-    Math_SmoothStepToF(&this->actor.scale.x, this->unk_15C, 0.1f, this->unk_15C * 0.1f, 0.0f);
+    Math_SmoothStepToF(&this->actor.scale.x, this->scale, 0.1f, this->scale * 0.1f, 0.0f);
+    temp = &D_80157D90;
+    
     this->actor.scale.z = this->actor.scale.x;
     this->actor.scale.y = this->actor.scale.x;
-
+    
     if (this->actor.gravity) {
         if (this->actor.bgCheckFlags & 0x0003) {
-            // Separate symbols seem to be used here for 0x80157D90 since the loads and stores are completely separate
-            if ((u32)D_80157D90 != globalCtx->gameplayFrames) {
-                D_80157D90_ = globalCtx->gameplayFrames;
-                D_80157D94 = 0;
-                for (i = 0; i < BG_ACTOR_MAX; i++) {
+            if (*temp != globalCtx->gameplayFrames) {
+                D_80157D90 = globalCtx->gameplayFrames;
+                D_80157D94[0] = 0;
+                for (i = 0; i < 50; i++) {
                     if (globalCtx->colCtx.dyna.bgActorFlags[i] & 1) {
                         dynaActor = globalCtx->colCtx.dyna.bgActors[i].actor;
                         if ((dynaActor != NULL) && (dynaActor->update != NULL) &&
                             ((dynaActor->world.pos.x != dynaActor->prevPos.x) ||
                              (dynaActor->world.pos.y != dynaActor->prevPos.y) ||
                              (dynaActor->world.pos.z != dynaActor->prevPos.z))) {
```

Comment:
cf my previous comments, does this not match as two ifs or do you like the single if better?

Reply:
fwiw i would prefer two ifs as well, the null checks on their own make sense. It should work id think